src/java.base/share/classes/sun/launcher/resources/launcher.properties

```diff
@@ -50,12 +50,14 @@ java.launcher.opt.footer = \
 \                  and ZIP archives to search for class files.\n\
 \    -p <module path>\n\
 \    --module-path <module path>...\n\
-\                  A {0} separated list of directories, each directory\n\
-\                  is a directory of modules.\n\
+\                  A {0} separated list of elements, each element is a file path\n\
+\                  to a module or a directory containing modules. Each module is either\n\
+\                  a modular JAR or an exploded-module directory.\n\
```

Comment:
JEP 261 uses "module definition" but that would be too hard to explain in the help output, so I think what you have is okay.